tests/php/ExternalLinks/ExternalLinksTest.php

```diff
@@ -125,6 +127,32 @@ public function testArchivedPagesAreHiddenFromReport()
         $this->assertEquals(3, BrokenExternalLinksReport::create()->sourceRecords()->count());
     }
 
+    public function testMissingPageGracefullyHandled(): void
+    {
+        // Run link checker to generate BrokenExternalPageTrackStatus
+        $task = CheckExternalLinksTask::create();
+        $task->runLinksCheck(PolyOutput::create(PolyOutput::FORMAT_ANSI, PolyOutput::VERBOSITY_QUIET));
+        // Simulate an old status that didn't complete, though was later reused
+        // also simluate a page it was tracking was deleted
+        $status = BrokenExternalPageTrackStatus::get_latest();
+        $status->Status = 'Running';
+        $status->write();
+        $pageID = BrokenExternalPageTrack::get()->max('ID') + 1;
+        $pageTrack = BrokenExternalPageTrack::create();
+        $pageTrack->PageID = $pageID;
+        $pageTrack->StatusID = $status->ID;
+        $pageTrack->write();
+        // Create an output buffer to capture the output
+        $bufferedOutput = new BufferedOutput();
+        $polyOutput = new PolyOutput(PolyOutput::FORMAT_ANSI, PolyOutput::VERBOSITY_NORMAL, false, $bufferedOutput);
+        // Run link checker to generate BrokenExternalPageTrackStatus
+        $task = new CheckExternalLinksTask();
```

Comment:
Why are we creating these again when we already had a `PolyOutput` and `CheckExternalLinksTask` created above?

Reply:
Updated